provider/provider.go

```diff
@@ -75,6 +76,34 @@ func OmitJsonArrayFields(val interface{}, omitList []interface{}) {
 	}
 }
 
+func timeSuffixToIntSec(tv string) int {
+	sz := len(tv)
+	l := sz - 1
+	if sz < 1 {
+		return 0
+	}
+	mult := 0
+	switch tv[sz-1] {
+	case 's':
+		mult = 1
+	case 'm':
+		mult = 60
+	case 'h':
+		mult = 60 * 60
+	case 'd':
+		mult = 60 * 60 * 24
+	default:
```

Comment:
Maybe guard against unwanted non-numerical characters?

Reply:
Note: this is handled by `err` possibly returned from `strconv.Atoi()` below.